app/src/androidTest/java/ch/epfl/skysync/screens/ConfirmFlightScreenTest.kt

```diff
@@ -49,23 +50,46 @@ class ConfirmFlightScreenTest {
   }
 
   @Test
-  fun backStackIsRightIfClickOnFlightDetailsThenFlightConfirm() {
+  fun backStackIsRightIfClickOnFlightDetailsThenFlightConfirm() = runTest {
+    val assignedFlight =
+        listOf(dbs.flight1, dbs.flight2, dbs.flight3, dbs.flight4).sortedBy { flight: Flight ->
+          flight.id
+        }
+    val retrievedFlights =
+        repository.flightTable.getAll(onError = { Assert.assertNull(it) }).sortedBy { flight: Flight
+          ->
+          flight.id
+        }
+    Assert.assertEquals(assignedFlight, retrievedFlights)
     composeTestRule.onNodeWithText("Home").performClick()
-    val nodes = composeTestRule.onAllNodesWithTag("flightCard")
-    for (i in 0 until nodes.fetchSemanticsNodes().size) {
-      nodes[i].performClick()
+
+    val canBeConfirmedFlights =
+        assignedFlight.filterIsInstance<PlannedFlight>().filter { it.readyToBeConfirmed() }
+
+    for (flight in canBeConfirmedFlights) {
+      composeTestRule
+          .onNodeWithTag("HomeLazyList")
+          .performScrollToNode(hasTestTag("flightCard${flight.id}"))
+      composeTestRule.onNodeWithTag("flightCard${flight.id}").performClick()
       var route = navController.currentBackStackEntry?.destination?.route
       Assert.assertEquals(Route.FLIGHT_DETAILS + "/{Flight ID}", route)
+
       composeTestRule.onNodeWithText("Confirm").performClick()
-      composeTestRule.waitForIdle()
-      route = navController.currentBackStackEntry?.destination?.route
-      Assert.assertEquals(Route.CONFIRM_FLIGHT + "/{Flight ID}", route)
-      composeTestRule
-          .onNodeWithTag("LazyList")
-          .performScrollToNode(hasText("Confirm"))
-          .assertIsDisplayed()
+
+      composeTestRule.waitUntil(2500) {
+        composeTestRule.onAllNodesWithText("Balloon").fetchSemanticsNodes().isNotEmpty()
+      }
+
+      composeTestRule.onNodeWithTag("LazyList").performScrollToNode(hasText("Enter Remark"))
```

Comment:
3rd copy of that code block between this test and the previous one, could've made a helper function to call when needed

Reply:
I don't think a function for one line is necessary.